subprojects/docs/src/docs/userguide/five_things.adoc

```diff
@@ -0,0 +1,125 @@
+
+Gradle is a flexible and powerful build tool that can easily feel intimidating when you start to learn it. There is potentially a lot to learn. However, there are some core principles that will make it easier to learn how to use Gradle effectively once you understand them. We present those principles here.
+
+== 1. Gradle is a general-purpose build tool
+
+Gradle allows you to build any software, because it makes few assumptions about what you're trying to build or how it should be done. The most notable restriction is that dependency management currently only supports Maven- and Ivy-compatible repositories and the filesystem.
+
+This doesn't mean you have to do a lot of work to create a build. Gradle makes it easy to build common types of project — say Java libraries — by adding a layer of conventions and prebuilt functionality through <<plugins#plugins,_plugins_>>. You can even create and publish custom plugins to encapsulate your own conventions and build functionality.
+
+== 2. The core model is based on a network of tasks
+
+Gradle models its builds as Directed Acyclic Graphs (DAGs) of tasks (units of work). What this means is that a build essentially configures a set of tasks and wires them together to create that DAG. Once the task graph has been created, Gradle determines which tasks need to be run in which order and then proceeds to execute them.
+
+This diagram shows two example task graphs, one abstract and the other concrete, with the dependencies between the tasks represented as arrows:
+
+.Two examples of a Gradle task graph
+image::task-dag-examples.png[Example task graphs]
+
+Almost any build process can be modeled as a graph of tasks in this way, which is one of the reasons why Gradle is so flexible.
+
+Tasks themselves consist of:
+
+ * Actions — pieces of work that do something, like copy files or compile source
+ * Inputs — values, files and directories that the actions use or operate on
+ * Outputs — files and directories that the actions modify or generate
+
+In fact, all of the above are optional depending on what the task needs to do. Some tasks — such as the <<base_plugin#sec:base_tasks,standard lifecycle tasks>> — don't even have any actions. They simply aggregate multiple tasks together as a convenience.
+
+NOTE: You choose which task to run. Save time by specifying the task that does what you need, but no more than that. If you just want to run the unit tests, choose the task that does that — typically `test`. If you want to package an application, most builds have an `assemble` task for that.
+
+One way to customize a build is to add extra steps to it. This is easy to do in Gradle: define your own tasks and insert them into the task graph via the <<tutorial_using_tasks#sec:task_dependencies,task dependency mechanism>>. You can have your tasks depend on any other task or configure existing tasks to depend on yours. The main restriction is that you cannot create any circular dependencies within the task graph.
```

Comment:
I'm a little hesitant to even include this paragraph. This is not a tutorial on how to write tasks and I want to keep it as focused as possible. It's just that I don't think it's obvious to people how to integrate their tasks into an existing task graph.

It's really a question of how important the information is to understanding how Gradle works. Talking about `doLast()` could end up adding an extra paragraph or two with little value to someone who knows little about Gradle at this point. Happy to be persuaded otherwise though.

Reply:
You may be able to remove this paragraph without losing anything critical, because I think it's implied and covered in docs linked to.

Reply:
@eriwen I've removed the paragraph, but added a new, similar statement to an earlier paragraph (the one starting "Almost any build process can be modeled ..."). I'm not sure about the exact wording, but I think it usefully highlights how a task graph comes into being.